### PR TITLE
Feature/remove old views

### PR DESCRIPTION
**Pull request by eheinrich**
Remove frontend code and templates that have been moved over to the frontend project. Kept and refactored tests that tested the views that were removed, but that are still testing useful things.

This PR also updates the querysets used for both `/api/requests/` and `/api/requestgroups/` to include `staff_view` and `view_authored_requests_only` in their filtering, the same as how those fields are taken into account for the observations querysets and the querysets that were used in the removed requestgroups pages.

### CODE REVIEW

File: observation_portal/observations/test/tests.py [hunk 1]
```diff
@@ -1640,6 +1648,148 @@ def test_with_one_obs_command_reports_dry_run_doesnt_modify_time(self):
         self.assertEqual(self.time_allocation.std_time_used, 0)
 
 
+class TestGetObservationsDetailAPIView(APITestCase):
```
Comment (eheinrich) [inline observation_portal/observations/test/tests.py:1651]: These tests, lines 1651 through 1792, are tests that I moved over from `observation_portal/observations/test_views.py` and refactored to test the observations API endpoints. Before they had been testing the `/observations` and `/observations/<obsId>` pages.

File: observation_portal/requestgroups/test/test_api.py [hunk 2]
```diff
@@ -112,13 +113,84 @@ def test_get_request_group_list_authenticated(self):
         result = self.client.get(reverse('api:request_groups-list'))
         self.assertContains(result, request_group.name)
 
-    def test_get_request_group_list_staff(self):
+    def test_get_request_group_list_is_staff_without_staff_view(self):
```
Comment (eheinrich) [inline observation_portal/requestgroups/test/test_api.py:116]: The new tests in this file are ones that I pulled over from `observation_portal/requestgroups/test/test_views.py` and refactored to test the requestgroups API endpoints.

File: observation_portal/requestgroups/viewsets.py [hunk 3]
```diff
@@ -47,12 +47,13 @@ def get_throttles(self):
         return super().get_throttles()
 
     def get_queryset(self):
-        if self.request.user.is_staff:
-            qs = RequestGroup.objects.all()
-        elif self.request.user.is_authenticated:
-            qs = RequestGroup.objects.filter(
-                proposal__in=self.request.user.proposal_set.all()
-            )
+        if self.request.user.is_authenticated:
```
Comment (eheinrich) [inline observation_portal/requestgroups/viewsets.py:50]: Updated queryset

File: observation_portal/requestgroups/viewsets.py [hunk 4]
```diff
@@ -217,15 +218,15 @@ class RequestViewSet(ListAsDictMixin, viewsets.ReadOnlyModelViewSet):
     ordering_fields = ('id', 'state')
 
     def get_queryset(self):
-        if self.request.user.is_staff:
-            qs = Request.objects.all()
-        elif self.request.user.is_authenticated:
-            qs = Request.objects.filter(
-                request_group__proposal__in=self.request.user.proposal_set.all()
-            )
+        if self.request.user.is_authenticated:
```
Comment (eheinrich) [inline observation_portal/requestgroups/viewsets.py:221]: Updated queryset

File: templates/base.html [hunk 5]
```diff
@@ -32,11 +32,11 @@
       </button>
       <div class="navbar-collapse collapse" id="navbarOptions">
         <ul class="navbar-nav ml-auto">
-          <li class="nav-item"><a class="nav-link" href="{% url 'requestgroups:list' %}">Home</a></li>
-          <li class="nav-item"><a class="nav-link" href="{% url 'requestgroups:create' %}">Submit Observation</a></li>
+          <li class="nav-item"><a class="nav-link" href="#">Home</a></li>
```
Comment (jnation3406) [inline templates/base.html:35]: This is just a placeholder since this file should go away later right?

Comment (eheinrich) [reply templates/base.html:35]: Yes that's right

**Pull request merged**